server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java

```diff
@@ -336,6 +345,45 @@ private List<JdbcGeneralResult> consumeStatement(Statement statement, SqlTuple s
         return executeResults;
     }
 
+    private void checkIsExternalTalbe(JdbcGeneralResult jdbcGeneralResult) {
+        if (dialectType == DialectType.OB_MYSQL || dialectType == DialectType.OB_ORACLE) {
+            String obVersion = ConnectionSessionUtil.getVersion(connectionSession);
+            if (VersionUtils.isGreaterThanOrEqualsTo(obVersion, support_External_Table_Version_Number_For_Ob)) {
+                JdbcColumnMetaData jdbcColumnMetaData = null;
+                try {
+                    jdbcColumnMetaData = jdbcGeneralResult.getQueryResult()
+                            .getMetaData().getColumns()
+                            .get(0);
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
```

Comment:
here may not throw exception, just regard as no external table, or the query will result in failed